rspace/src/main/scala/coop/rchain/rspace/ReportingRspace.scala

```diff
@@ -143,7 +135,7 @@ class ReportingRspace[F[_]: Concurrent: ContextShift: Log: Metrics: Span, C, P,
       reportingConsume  = ReportingConsume(channels, wk.patterns, wk.continuation, wk.peeks.toSeq)
       reportingProduces = dataCandidates.map(dc => ReportingProduce(dc.channel, dc.datum.a))
       _ <- Sync[F].delay(
-            softReport.update(s => s :+ ReportingComm(reportingConsume, reportingProduces))
+            softReport.update(_ :+ ReportingComm(reportingConsume, reportingProduces))
```

Comment:
What is the type of `softReport.update(_ :+ ReportingComm(reportingConsume, reportingProduces))`?

What type `Sync[F].delay` accepts?

To work with these situations you can ask IntelliJ to help you.
Assign a name to the expression you are interested in and hover with a mouse for IntelliJ to show you the type.
```scala
whatIsTheType = Sync[F].delay(
  softReport.update(_ :+ ReportingComm(reportingConsume, reportingProduces))
)
```
Note: `<-` in for..yield (sugar for flatMap) can _peel of_ one level of nesting.

Reply:
That awkward moment when you see the code you wrote a week ago...
I fixed this and other same mistakes in 3 places of this file